mongodb_store/scripts/mongodb_play.py

```diff
@@ -172,6 +173,13 @@ def run(self, running):
                     delay = publish_time - now
                     rospy.sleep(delay)
 
+                # if message is an action goal then the goal_id time stamp should be changed to the current wall clock time
+                if self.msg_contains_goal_id is not None:
+                    float_secs = time.time()
+                    secs = int(float_secs)
+                    msg.goal_id.stamp.secs = secs
+                    msg.goal_id.stamp.nsecs = int((float_secs - secs) * 1000000000)
```

Comment:
I think you can use `msg.goal_id.stamp = ` `rospy.Time.now()` or `rospy.get_rostime()`?

doc: http://wiki.ros.org/rospy/Overview/Time#Getting_the_current_time

```
now = rospy.get_rostime()
rospy.loginfo("Current time %i %i", now.secs, now.nsecs)
```

Reply:
I suggested `msg.goal_id.stamp = 0`, but not sure if it works...

Reply:
That returns the current sim time. This is what the definition of what Time.now() says: 
```
Create new L{Time} instance representing current time. This 
140          can either be wall-clock time or a simulated clock. It is 
141          strongly recommended that you use the now() factory to create 
142          current time representations instead of reading wall-clock 
143          time and create Time instances from it. 
144           
145          @return: L{Time} instance for current time 
146          @rtype: L{Time} 
```
Time.now() will return the simulated clock time, which is different than the wall clock time. If we set the goal_id stamp to the simulated clock time it'll be the same value as before. The wallclock time will allow for no issues.